src/com/ms/silverking/io/StreamParser.java

```diff
@@ -18,177 +16,182 @@
+  }
+
+  public static List<String> parseLines(InputStream inStream) throws IOException {
+    return parseLines(inStream, TrimMode.trim, CloseMode.close, null);
+  }
+
+  public static List<String> parseLines(InputStream inStream, String regex) throws IOException {
+    return parseLines(inStream, TrimMode.trim, CloseMode.close, regex);
+  }
+
+  public static List<String> parseLines(InputStream inStream, CloseMode closeMode) throws IOException {
+    return parseLines(inStream, TrimMode.trim, closeMode, null);
+  }
+
+  public static List<String> parseLines(InputStream inStream, TrimMode trimMode) throws IOException {
+    return parseLines(inStream, trimMode, CloseMode.close, null);
+  }
+
+  public static List<String> parseLines(InputStream inStream, TrimMode trimMode, CloseMode closeMode, String regex)
+      throws IOException {
+    BufferedReader reader;
+    ArrayList<String> lines;
+    String line;
+
+    reader = new BufferedReader(new InputStreamReader(inStream));
+    try {
+      lines = new ArrayList<String>();
+      do {
+        line = reader.readLine();
+        if (line != null) {
+          if (regex == null || line.matches(regex)) {
             if (trimMode == TrimMode.trim) {
-                return reader.readLine().trim();
+              lines.add(line.trim());
             } else {
-                return reader.readLine();
-            }
-        } finally {
-            if (closeMode == CloseMode.close) {
-                reader.close();
+              lines.add(line);
             }
+          }
         }
-    }    
-    
-    public static Map<String,String> parseMap(InputStream inStream) throws IOException {
-        return parseMap(inStream, TrimMode.trim, CloseMode.close);
-    }
-    
-    public static Map<String,String> parseMap(InputStream inStream, TrimMode trimMode, CloseMode closeMode) throws IOException {
-        Map<String,String>  map;
-        List<String>        lines;
-        
-        map = new HashMap<>();
-        lines = parseLines(inStream, trimMode, closeMode, null);
-        for (String line : lines) {
-            int mapCharIndex;
-            
-            mapCharIndex = line.indexOf(mapChar);
-            if (mapCharIndex > 0) {
-                if (line.indexOf(mapChar, mapCharIndex + 1) >= 0) {
-                    throw new IOException("Bad map entry: "+ line);
-                }
-                map.put(line.substring(0, mapCharIndex), line.substring(mapCharIndex + 1));
-            }
+      } while (line != null);
+    } finally {
+      if (closeMode == CloseMode.close) {
+        reader.close();
+      }
+    }
+    return lines;
+  }
+
+  public static String parseLine(File file) throws IOException {
+    return parseLine(file, TrimMode.trim, CloseMode.close);
+  }
+
+  public static String parseLine(File file, TrimMode trimMode, CloseMode closeMode) throws IOException {
+    return parseLine(new FileInputStream(file), trimMode, closeMode);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when StreamParser.parseLine throws an exception. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).